File: common/src/main/java/io/github/slimeymc/slimeys_utility/integration/SlimeysUtilityKubeJSPlugin.java
package io.github.slimeymc.slimeys_utility.integration;

import dev.latvian.mods.kubejs.script.BindingsEvent;
import io.github.slimeymc.slimeys_utility.physicify.ShipPhysicifyKt;
import io.github.slimeymc.slimeys_utility.util.VectorConversionKt;


public class SlimeysUtilityKubeJSPlugin {
    public static void addBindings(BindingsEvent event) {
        // Vector Conversion
        event.add("VectorConversionsExt", VectorConversionKt.class);

        // ShipPhysicify
        event.add("ShipPhysicify", ShipPhysicifyKt.class);
    }
}

File: common/src/main/kotlin/io/github/slimeymc/slimeys_utility/integrations/SlimeysUtilityKubeJSPlugin.kt
package io.github.slimeymc.slimeys_utility.integrations

import dev.latvian.mods.kubejs.KubeJSPlugin
import dev.latvian.mods.kubejs.script.BindingsEvent
import io.github.slimeymc.slimeys_utility.items.*
import io.github.slimeymc.slimeys_utility.ship.ShipData
import io.github.slimeymc.slimeys_utility.util.RelocateLevel

class SlimeysUtilityKubeJSPlugin: KubeJSPlugin() {
    override fun addBindings(event: BindingsEvent) {
        // Utility stuff
        event.add("ShipData", ShipData::class.java)
        event.add("RelocateLevel", RelocateLevel::class.java)

        // Item
        event.add("ShipSelectingItem", ShipSelectionItem::class.java)
        event.add("ShipAssemblerItem", ShipAssemblerItem::class.java)
        event.add("ShipScalerItem", ShipScalerItem::class.java)
        event.add("ShipTaggerItem", ShipTaggerItem::class.java)
    }
}